include/librealsense2/dds/dds-topic-reader.cpp

```diff
@@ -0,0 +1,109 @@
+    history().depth = 10;  // default is 1
+
+    // We don't want to miss connection/disconnection events
+    // (default is best-effort)
+    reliability().kind = eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+
+    // The Subscriber receives samples from the moment it comes online, not before
+    // (this is the default, but bears repeating)
+    durability().kind = eprosima::fastdds::dds::VOLATILE_DURABILITY_QOS;
+
+    // The writer-reader handshake is always done on UDP, even when data_sharing (shared memory) is used for
+    // actual messaging. This means it is possible to send a message and receive it on the reader's
+    // side even before the UDP handshake is complete:
+    // 
+    //   1. The writer goes up and broadcasts its presence periodically; no readers exist
+    //   2. The reader joins and broadcasts its presence, again periodically; it doesn<92>t know about the writer yet
+    //   3. The writer sees the reader (in-between broadcasts) so sends a message
+    //   4. The reader gets the message and discards it because it does not yet recognize the writer
+    // 
+    // This depends on timing. When shared memory is on, step 3 is so fast that this miscommunication is much more
+    // likely. This is a known gap in the DDS standard.
+    //
+    // We can either insert a sleep between writer creation and message sending or we can disable
+    // data_sharing for this topic, which we did here.
+    // 
+    // See https://github.com/eProsima/Fast-DDS/issues/2641
+    //
+    data_sharing().off();
+}
+
+
+void dds_topic_reader::run( reader_qos const & rqos )
+{
+    // The Subscriber manages the activities of several DataReader entities
+    _subscriber = DDS_API_CALL(
+        _topic->get_participant()->get()->create_subscriber( eprosima::fastdds::dds::SUBSCRIBER_QOS_DEFAULT ) );
+
+    // The DataReader is the Entity used by the application to subscribe to updated values of the data
+    eprosima::fastdds::dds::StatusMask status_mask;
+    if( _on_subscription_matched )
+        status_mask << eprosima::fastdds::dds::StatusMask::subscription_matched();
+    if( _on_data_available )
+        status_mask << eprosima::fastdds::dds::StatusMask::data_available();
+    _reader = DDS_API_CALL(
+        _subscriber->create_datareader( _topic->get(), rqos, status_mask.any() ? this : nullptr, status_mask ) );
+}
+
+
+void dds_topic_reader::on_subscription_matched(
+    eprosima::fastdds::dds::DataReader *, eprosima::fastdds::dds::SubscriptionMatchedStatus const & info )
+{
+    // Called when the subscriber is matched (un)with a Writer
+    if( _on_subscription_matched )
+        _on_subscription_matched( info );
+}
+
+void dds_topic_reader::on_data_available( eprosima::fastdds::dds::DataReader * )
+{
+    // Called when a new  Data Message is received
+    if( _on_data_available )
+        _on_data_available();
+}
+
+
+}  // namespace dds
+}  // namespace librealsense
```

Comment:
Missing EOL

Reply:
It's not missing, just not there. It's fine...

Reply:
I read about it online and I think if you read this you will agree it's better to have it:
```
It's not just bad style, it can lead to unexpected behavior when using other tools on the file.

Here is test.txt:

first line
second line
There is no newline character on the last line. Let's see how many lines are in the file:

$ wc -l test.txt
1 test.txt
Maybe that's what you want, but in most cases you'd probably expect there to be 2 lines in the file.

Also, if you wanted to combine files it may not behave the way you'd expect:

$ cat test.txt test.txt
first line
second linefirst line
second line
Finally, it would make your diffs slightly more noisy if you were to add a new line. If you added a third line, it would show an edit to the second line as well as the new addition.
```

Reply:
I'm sorry, but just because `wc` has a bug in it doesn't mean we need to jump thru hoops.
There's nothing wrong with the source...

Reply:
And the point about "bad style"?! Give me a break.